vignettes/wflow-06-gitlab.Rmd

```diff
@@ -0,0 +1,147 @@
+---
+title: "Hosting worklowr using GitLab"
+subtitle: "workflowr version `r utils::packageVersion('workflowr')`"
+author: "John Blischak"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Frequently asked questions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## What is in this vignette?
+
+By default workflowr assumes that the project will be hosted on GitHub, but this
+is not always the case. Users may prefer to use another service or have a
+private git repository hosting instance. This vignette details some of the steps
+required to host a workflowr project on GitLab. Similar steps will be required
+for other platforms but some of the specifics will be different.
+
+As these steps are not currently integrated into workflowr this vignette assumes
+a basic knowledge of using git from a command line interface.
+
+## Step 0: Set up a project
+
+The first thing we need to do is set up the project we want to host. We can do
+this by following the first few steps of the instructions in the
+"Getting started" vignette, however we can't use the `wflow_git_remote`
+command as it assumes we will be using GitHub. When you get to that stage follow
+the rest of the steps in this vignette.
+
+## Step 1: Set up a remote repository
+
+Log in to the GitLab instance you want to use and create a repository to host
+your project. You will need to know your user name and the repository name for
+the following steps (here we are going to use "myname" and "myproject") as well
+as a URL for the hosting instance.
+
+We now want to connect our local git repository to the one we have just created
+on GitLab. Helpully GitLab provides us instructions on how to do that under
+"Existing Git respository" on our new repository page. The simplest thing to
+do is copy these commands and run them in your local repository directory. If
+you can't see the commands or want to do it manually the most important parts
+are:
+
+```
+git remote add origin HOSTING_URL/myname/myproject.git
+git push origin master
+```
+
+If this step has worked correctly you should be able to refresh your GitLab
+page and see all the files in your workflowr project.
+
+## Step 2: Change the build directory
+
+By default workflowr stores produced HTML files in the `docs` directory but
+GitLab can only host files that are in a directory called `public`. To fix this
+we need to tell workflowr to change the directory name by editing the
+`_site.yml` file in the `analysis` directory. We want to change the line that
+says `output_dir: "../docs"` to `output_dir: "../public"`. The command below
+uses `sed` to do this but you could also do it in RStudio or your favourite text
+editor.
+
+```
+sed s/docs/public/ analysis/_site.yml
```

Comment:
This `sed` command doesn't work because the [code I wrote](https://github.com/jdblischak/workflowr/issues/42#issuecomment-408139202) was wrong (I'll go back to that Issue and correct it). It needs the `-i` flag to edit the file in place:

```
# Unix
sed -i s/docs/public/ analysis/_site.yml
# Unix or macOS
sed -i'.bk' s/docs/public/ analysis/_site.yml
```

Reply:
Yeah I noticed that when I tried to set up a new project but I'm not really familiar with sed so didn't know how to fix it. I think most people will just use a text editor anyway but it's good to have an example command in there.

Reply:
Great! See 048897f28c519d25a67c199e5dab5212ac947cb2